specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/dedicatedhsm.json

```diff
@@ -0,0 +1,763 @@
+          "description": "This field will be used when RP does not support Availability zones."
+        },
+        "statusMessage": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Status Message."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Provisioning state.",
+          "enum": [
+            "Succeeded",
+            "Provisioning",
+            "Allocating",
+            "Connecting",
+            "Failed",
+            "CheckingQuota",
+            "Deleting"
+          ],
+          "x-ms-enum": {
+            "name": "JsonWebKeyType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Succeeded",
+                "description": "The dedicated HSM has been full provisioned."
+              },
+              {
+                "value": "Provisioning",
+                "description": "The dedicated HSM is currently being provisioned."
+              },
+              {
+                "value": "Allocating",
+                "description": "A device is currently being allocated for the dedicated HSM resource."
+              },
+              {
+                "value": "Connecting",
+                "description": "The dedicated HSM is being connected to the virtual network."
+              },
+              {
+                "value": "Failed",
+                "description": "Provisioning of the dedicated HSM has failed."
+              },
+              {
+                "value": "CheckingQuota",
+                "description": "Validating the subscription has sufficient quota to allocate a dedicated HSM device."
+              },
+              {
+                "value": "Deleting",
+                "description": "The dedicated HSM is currently being deleted."
+              }
+            ]
+          }
+        }
+      },
+      "description": "Properties of the dedicated hsm",
+      "type": "object"
+    },
+    "DedicatedHsm": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "SKU details"
+        },
+        "zones": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The Dedicated Hsm zones."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DedicatedHsmProperties",
+          "description": "Properties of the dedicated HSM"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "required": [
+        "location",
+        "sku",
+        "properties"
+      ],
+      "description": "Resource information with extended details.",
+      "type": "object"
+    },
+    "DedicatedHsmPatchParameters": {
```

Comment:
>DedicatedHsmPatchParameters [](http://example.com/codeflow?start=5&length=27)

Can you not patch with the SKU to update the sku?

Reply:
Can you explain again ?

Reply:
I mean, is SKU update not a supported operation?

Reply:
SKU is supported, but for this service is only supported via ARM client/Invoke-RestMethod. Please see documentation [here](https://learn.microsoft.com/en-us/azure/payment-hsm/change-performance-level). 
Update-azdedicatedHsm command does not include SKU, documentation [here](https://learn.microsoft.com/en-us/powershell/module/az.dedicatedhsm/update-azdedicatedhsm?view=azps-12.0.0)

Reply:
>but for this service is only supported via ARM client/Invoke-RestMethod. Please see documentation here. 

That's still supporting it via this same REST API then, right? Swagger is supposed to document the whole REST API, not just the bits of it that the SDKs and powershell make easy to use